blocks/market-insight/market-insight.js

```diff
@@ -84,35 +56,53 @@ function addSocialButtonEvent(button, block) {
   });
 }
 
-async function decorateCards(block, placeholders, previousNode) {
-  const results = await fetchMarketInsightMockData();
+async function decorateCards(block, placeholders, cardCount, previousNode) {
+  const queryObj = await fetchMarketInsightMockData();
+  const results = queryObj.map((el) => {
+    const elArr = el.PublishedOnDate.split(' ');
+    const publishDate = elArr[0];
+    const publishTime = elArr[1];
+    const formatPublishDate = publishDate.split('-').reverse();
+    el.PublishedOnDate = `${formatPublishDate} ${publishTime}`;
+    return el;
+  }).sort((a, b) => {
+    const dateA = new Date(a.PublishedOnDate);
+    const dateB = new Date(b.PublishedOnDate);
+    return dateB - dateA;
+  });
+
   const powerBy = (placeholders.powerby ?? '').trim();
   const publishedOn = (placeholders.publishedon ?? '').trim();
   const ul = createElement('ul', '');
-  // show 3 cards by default
-  for (let index = 0; index < (results.length > 3 ? 3 : results.length); index += 1) {
+  const loopNum = results.length > cardCount ? cardCount : results.length;
+  for (let index = 0; index < loopNum; index += 1) {
     const result = results[index];
     const li = createElement('li', '');
     const title = createElement('div', 'cards-title');
     const description = createElement('div', 'cards-description');
     const powerby = createElement('div', 'cards-powerby');
     // Cards title
     const h3 = createElement('h3', '');
-    const aLink = createElement('a', '');
-    aLink.href = result.link;
-    aLink.target = '_blank';
-    aLink.append(result.title);
-    h3.append(aLink);
+    const titleContent = result.Title ?? '';
+    if (result.PermLink) {
+      const aLink = createElement('a', '');
+      aLink.href = result.PermLink;
+      aLink.target = '_blank';
+      aLink.append(titleContent);
+      h3.append(aLink);
+    } else {
+      h3.textContent = titleContent;
+    }
     title.append(h3);
     // Cards description
-    description.innerHTML = decodeURIComponent(result.description);
+    description.innerHTML = decodeURIComponent(result.ShortDescription ?? '');
     // Cards powerby
     const powerbyDiv = createElement('div', '');
     const powerbyContent = createElement('p', '');
     powerbyContent.textContent = powerBy;
     const publishedOnContent = createElement('p', '');
-    const publishedon = result.publishedon.replaceAll(' ', '-');
-    publishedOnContent.textContent = `${publishedOn} ${publishedon}`;
+    const publishOndate = result.PublishedOn ? result.PublishedOn.replaceAll(' ', '-') : '';
```

Comment:
I am thinking if we should use [some standard date formatting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat) function here.

Reply:
I thought the same, but the original site shows like this. I kept the same look&feel.

<img width="457" alt="image" src="https://github.com/aemsites/icicidirect/assets/57093988/86c99d4e-11d8-424a-9c20-f5d5dd090414">